packages/gatsby-plugin-mdx/src/gatsby-node.ts

```diff
@@ -0,0 +1,303 @@
+ */
+export const resolvableExtensions: GatsbyNode["resolvableExtensions"] = (
+  _data,
+  pluginOptions: IMdxPluginOptions
+) => defaultOptions(pluginOptions).extensions
+
+/**
+ * Convert MDX to JSX so that Gatsby can extract the GraphQL queries and render the pages.
+ */
+export const preprocessSource: GatsbyNode["preprocessSource"] = async (
+  { filename, contents, getNode, getNodesByType, pathPrefix, reporter, cache },
+  pluginOptions: IMdxPluginOptions
+) => {
+  const { extensions } = defaultOptions(pluginOptions)
+
+  const mdxOptions = await enhanceMdxOptions(pluginOptions, {
+    getNode,
+    getNodesByType,
+    pathPrefix,
+    reporter,
+    cache,
+  })
+
+  const ext = path.extname(filename)
+
+  if (!extensions.includes(ext)) {
+    return undefined
+  }
+
+  const code = await compileMDX(
+    contents,
+    {
+      id: ``,
+      children: [],
+      parent: ``,
+      internal: { contentDigest: ``, owner: ``, type: `` },
+    },
+    {
+      id: ``,
+      children: [],
+      parent: ``,
+      internal: { contentDigest: ``, owner: ``, type: `` },
+      absolutePath: filename,
+      sourceInstanceName: `mocked`,
+    },
+    mdxOptions
+  )
+
+  return code.toString()
+}
+
+export const createSchemaCustomization: GatsbyNode["createSchemaCustomization"] =
+  async ({ actions, schema }) => {
+    const { createTypes } = actions
+    const typeDefs = [
+      schema.buildObjectType({
+        name: `MdxFrontmatter`,
+        fields: {
+          title: `String`,
+        },
+      }),
+      schema.buildObjectType({
+        name: `Mdx`,
+        fields: {
+          rawBody: `String!`,
+          frontmatter: `MdxFrontmatter!`,
+          slug: `String`,
+          title: `String`,
+        },
+        interfaces: [`Node`],
+      }),
+    ]
+    createTypes(typeDefs)
+  }
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const unstable_shouldOnCreateNode: GatsbyNode["unstable_shouldOnCreateNode"] =
+  ({ node }: { node: FileSystemNode }, pluginOptions: IMdxPluginOptions) => {
+    const { extensions } = defaultOptions(pluginOptions)
+    return node.internal.type === `File` && extensions.includes(node.ext)
+  }
+
+/**
+ * Create Mdx nodes from MDX files.
+ */
+export const onCreateNode: GatsbyNode<FileSystemNode>["onCreateNode"] = async ({
+  node,
+  loadNodeContent,
+  actions: { createNode, createParentChildLink },
+  createNodeId,
+}) => {
+  const content = await loadNodeContent(node)
+
+  const { data: frontmatter } = grayMatter(content)
+
+  // Use slug from frontmatter, otherwise fall back to the file name and path
+  const slug =
+    frontmatter.slug ||
+    [node.relativeDirectory, node.name === `index` ? `` : node.name]
+      .filter(Boolean)
+      .join(`/`)
+
+  // Use title from frontmatter, otherwise fall back to the file name
+  const title =
+    frontmatter.title || node.name === `index`
+      ? `Home`
+      : sentenceCase(node.name)
```

Comment:
The old plugin didn't have that, why did we add that?

It also doesn't seem to work correctly in my site:

![image](https://user-images.githubusercontent.com/16143594/171989051-2a01440b-6260-4290-8068-6cc98603cbf6.png)

Reply:
I adjusted the logic and removed the assumption we can use `Home` for index files